### ISSUE TITLE
What is the recommended mechanism to ignore actions from Swagger with minimal actions?

### ISSUE THREAD

**Issue by martincostello** [dotnet/aspnetcore#34061]
I've been playing around with using minimal actions to build a sample app that's a mix of API, Identity and Razor Pages, but exclusively using minimal actions to implement any endpoints in the app, not just the API. I'm currently using this with a daily build of the preview 7 SDK (`6.0.100-preview.7.21330.1`).

When plugging-in Swagger with the minimal configuration, all the minimal actions endpoints are returned in the Swagger documentation.

I've manually hidden things with the following code:

```csharp
builder.Services.AddSwaggerGen(options =>
{
    options.SwaggerDoc("v1", new OpenApiInfo { Title = "Todo API", Version = "v1" });

    options.DocInclusionPredicate(
        (_, description) => description.RelativePath?.StartsWith("api/", StringComparison.Ordinal) == true);
});
```

However what I would have _liked_ to have done instead is to explicitly hide the actions using the `ApiExplorerSettings` attribute using the new support for attributes on lambdas, that doesn't seem to be working.

This is what I tried:

```csharp
builder.MapPost("/signin", [ApiExplorerSettings(IgnoreApi = true)] () =>
    Results.Challenge(
        new AuthenticationProperties { RedirectUri = "/" },
        GitHubAuthenticationDefaults.AuthenticationScheme));
```

Another approach I've used in the past is the `IActionModelConvention` support, but that's part of MVC so doesn't appear to be valid either.

While the `DocInclusionPredicate()` approach works with a simple app, it doesn't feel like it would scale well from a maintenance point of view, particularly if the naming/paths weren't following a specific convention. It also doesn't _seem_ possible to query for any decorated attributes (i.e. the `ApiExplorerSettingsAttribute`) from the `ApiDescription` passed to the method.

What is/will be the intended mechanism developers using minimal actions would use to hide specific actions from the API Explorer for use cases like Swagger?

**Comment by martincostello** [comment]
It's possible that using `[ApiExplorerSettings]` _should_ work, but doesn't because the metadata on the endpoint isn't copied over.

I've opened #34065 as a draft that resolves that, assuming it's the correct thing to do.

**Comment by martincostello** [comment]
Having dug through how this works today for MVC, it looks like this is currently dealt with by [`ApplicationModel`](https://github.com/dotnet/aspnetcore/blob/1b819d05de41561cc974af066737c38ded877ef7/src/Mvc/Mvc.Core/src/ApplicationModels/ApplicationModel.cs#L38) with the `IApiExplorerModel` implementation and the way that [`DefaultApplicationModelProvider`](https://github.com/dotnet/aspnetcore/blob/1b819d05de41561cc974af066737c38ded877ef7/src/Mvc/Mvc.Core/src/ApplicationModels/DefaultApplicationModelProvider.cs#L19) implements [`IApplicationModelProvider`](https://github.com/dotnet/aspnetcore/blob/1b819d05de41561cc974af066737c38ded877ef7/src/Mvc/Mvc.Core/src/ApplicationModels/IApplicationModelProvider.cs#L9) which eventually goes through [`ApplicationModelConventions.ApplyConventions()`](https://github.com/dotnet/aspnetcore/blob/3c7a5e7563a66c7f9f9ec04816e5a74aeabcec81/src/Mvc/Mvc.Core/src/ApplicationModels/ApplicationModelConventions.cs#L20) to apply the built-in conventions that then use the `IApiExplorerModel`s to apply all of the built-in `IApplicationModelConvention` implementations, which eventually result in things being suppressed when `[ApiExplorerSettings(IgnoreApi = true)]` is used.

`ApplicationModel` seems to be controller-centric though, so making it aware of arbitrary endpoints seems wrong, but not doing so means that all of the built-in conventions can't be applied to minimal actions.

The model doesn't support having multiple models either from what I can tell, so it doesn't look like having a parallel implementation that didn't care about controllers and operated in the registered endpoints would make much sense either.

**TL;DR** - It looks like #34065 wouldn't make `[ApiExplorerSettings]` Just Work™️, but it would make it possible for consuming applications to hide it manually with `DocInclusionPredicate()` when using Swashbuckle.AspNetCore. I'm not sure what the It Just Works™️ approach would need to be for minimal actions.

**Comment by martincostello** [comment]
Resolved by #34906.
